### PR TITLE
TravisCI: test on arm64, ppc64le, and s390x

### PR DESCRIPTION
**Pull request by mr-c**
This pull request will help debug  #2388

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [x] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -62,30 +62,75 @@ matrix:
...
     - stage: test
       python: pypy3
       env: TOXENV=pypy3-nocov
+      addons:
+        apt:
+          packages:
+            - *amd64_only_packages
+            - *default_packages
 
-sudo: false
```
Comment (mr-c) [inline .travis.yml]: Deprecated: https://blog.travis-ci.com/2018-11-19-required-linux-infrastructure-migration

**Comment by mr-c** [comment]
Success! :-D

[s390x errors](https://travis-ci.org/biopython/biopython/jobs/624955810#L850), which are similar [but not the same as the s390x errors we see in Debian](https://buildd.debian.org/status/fetch.php?pkg=python-biopython&arch=s390x&ver=1.75%2Bdfsg-1&stamp=1575482647&raw=0)

Similar for [arm64](https://travis-ci.org/biopython/biopython/jobs/624955808#L850) and [ppc64le](https://travis-ci.org/biopython/biopython/jobs/624955809#L816)

```
======================================================================

ERROR: test_matrix_correlations (test_SubsMat.TestGeo)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/build/biopython/biopython/Tests/test_SubsMat.py", line 1511, in test_matrix_correlations

    correlation = SubsMat.two_mat_correlation(blosum30, blosum90)

  File "/home/travis/build/biopython/biopython/build/lib.linux-s390x-3.8/Bio/SubsMat/__init__.py", line 640, in two_mat_correlation

    correlation_matrix = numpy.corrcoef(values, rowvar=0)

  File "<__array_function__ internals>", line 5, in corrcoef

  File "/home/travis/build/biopython/biopython/.tox/py38-cover/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2526, in corrcoef

    c = cov(x, y, rowvar)

  File "<__array_function__ internals>", line 5, in cov

  File "/home/travis/build/biopython/biopython/.tox/py38-cover/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2381, in cov

    X = array(m, ndmin=2, dtype=dtype)

ValueError: ndmin bigger than allowable number of dimensions NPY_MAXDIMS (=32)

======================================================================

ERROR: test_doctests (test_Tutorial.TutorialTestCase)

Run tutorial doctests.

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/build/biopython/biopython/Tests/test_Tutorial.py", line 259, in test_doctests

    raise ValueError("%i Tutorial doctests failed: %s" %

ValueError: 2 Tutorial doctests failed: test_chapter_align_line_02303, test_chapter_align_line_02371

======================================================================

ERROR: test_three_letter_amino_acids_global (test_pairwise_aligner.TestSequencesAsLists)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/build/biopython/biopython/Tests/test_pairwise_aligner.py", line 1785, in test_three_letter_amino_acids_global

    score = aligner.score(seq1, seq2)

  File "/home/travis/build/biopython/biopython/build/lib.linux-s390x-3.8/Bio/Align/__init__.py", line 1527, in score

    return _aligners.PairwiseAligner.score(self, seqA, seqB)

ValueError: sequence has unexpected format

======================================================================

ERROR: test_three_letter_amino_acids_local (test_pairwise_aligner.TestSequencesAsLists)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/build/biopython/biopython/Tests/test_pairwise_aligner.py", line 1830, in test_three_letter_amino_acids_local

    score = aligner.score(seq1, seq2)

  File "/home/travis/build/biopython/biopython/build/lib.linux-s390x-3.8/Bio/Align/__init__.py", line 1527, in score

    return _aligners.PairwiseAligner.score(self, seqA, seqB)

ValueError: sequence has unexpected format

======================================================================

FAIL: test_aligner_string_errors (test_pairwise_aligner.TestArgumentErrors)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/build/biopython/biopython/Tests/test_pairwise_aligner.py", line 1861, in test_aligner_string_errors

    aligner.score("AAA", "AA&")

AssertionError: ValueError not raised
```

**Comment by peterjc** [comment]
Good idea - I was meaning to try/suggest this. Can you mark the new entries as allowed failures in the short term?

**Comment by mdehoon** [comment]
For the pairwise aligner, the bug on arm64 is due to `char` being `unsigned char`, whereas on other systems it's `signed char`, which is what the code expects. This will be a simple fix.

**Comment by mr-c** [comment]
Cool, now that #2402 is merged we're down to a [single test failure](https://travis-ci.org/biopython/biopython/jobs/625300915#L845) on a single architecture (s390x):

```
======================================================================
ERROR: test_matrix_correlations (test_SubsMat.TestGeo)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/biopython/biopython/Tests/test_SubsMat.py", line 1511, in test_matrix_correlations
    correlation = SubsMat.two_mat_correlation(blosum30, blosum90)
  File "/home/travis/build/biopython/biopython/build/lib.linux-s390x-3.8/Bio/SubsMat/__init__.py", line 640, in two_mat_correlation
    correlation_matrix = numpy.corrcoef(values, rowvar=0)
  File "<__array_function__ internals>", line 5, in corrcoef
  File "/home/travis/build/biopython/biopython/.tox/py38-cover/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2526, in corrcoef
    c = cov(x, y, rowvar)
  File "<__array_function__ internals>", line 5, in cov
  File "/home/travis/build/biopython/biopython/.tox/py38-cover/lib/python3.8/site-packages/numpy/lib/function_base.py", line 2381, in cov
    X = array(m, ndmin=2, dtype=dtype)
ValueError: ndmin bigger than allowable number of dimensions NPY_MAXDIMS (=32)
```

**Comment by mr-c** [comment]
@peterjc The sole remaining failing test is allowed to be skipped if it fails in that circumstance (s390x) and thus the build is green :-)

**Comment by peterjc** [comment]
Ah - there's a conflict since I just merged your phyml change - would you mind doing a ``git rebase`` or ``git merge`` and then I will squash merge this to the master?

**Comment by mr-c** [comment]
@peterjc No problem. I've rebase and force pushed.

**Comment by mr-c** [comment]
@peterjc Yeah, I only learned of it from other group's configurations and web searching related to this issue. The `&` and `*` are YAML references and anchors: https://yaml.org/refcard.html

**Comment by peterjc** [comment]
Thank you!

**Comment by peterjc** [comment]
Unfortunately I've had to mark ARM64 and ppc64le as allowed failures on Travis as they keep timing out.

I suspect part of the issue the there are no pre-compiled wheels for numpy on these architectures, but even if we wanted to test without numpy (see #2418) it would be counter productive as most of the C code in Biopython uses numpy.

**Comment by mr-c** [comment]
@peterjc Shouldn't the numpy wheels get cached automatically?

**Comment by peterjc** [comment]
There seem to be some cache issues on the new architectures.

Strangely the builds on https://travis-ci.org/peterjc/biopython/builds seem to work better than on https://travis-ci.org/biopython/biopython/builds which might be explained by my personal account having managed to build and cache numpy once?

**Pull request merged**